From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Justin M. Forbes" <jforbes@fedoraproject.org>
Date: Thu, 10 Sep 2020 16:39:54 -0500
Subject: [PATCH] Filter out LTO build options from the perl ccopts

Kernel tools currently does not build with LTO options enabled.  We can
turn off the option in most places through the spec file, but for the
perl bindings, it auto detects what perl was built with and uses those.
This patch will filter out the lto options allowing kernel-tools
packages to build correctly.  I expect this patch to go away when the
LTO issues are worked out upstream.

Upstream Status: Red Hat only

Signed-off-by: Justin M. Forbes <jforbes@fedoraproject.org>
---
 tools/perf/Makefile.config | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/perf/Makefile.config b/tools/perf/Makefile.config
index 190be4fa5c21..28beab03575a 100644
--- a/tools/perf/Makefile.config
+++ b/tools/perf/Makefile.config
@@ -748,6 +748,7 @@ else
   PERL_EMBED_LIBADD = $(call grep-libs,$(PERL_EMBED_LDOPTS))
   PERL_EMBED_CCOPTS = $(shell perl -MExtUtils::Embed -e ccopts 2>/dev/null)
   PERL_EMBED_CCOPTS := $(filter-out -specs=%,$(PERL_EMBED_CCOPTS))
+  PERL_EMBED_CCOPTS := $(filter-out -flto=auto -ffat-lto-objects, $(PERL_EMBED_CCOPTS))
   PERL_EMBED_LDOPTS := $(filter-out -specs=%,$(PERL_EMBED_LDOPTS))
   FLAGS_PERL_EMBED=$(PERL_EMBED_CCOPTS) $(PERL_EMBED_LDOPTS)

-- 
2.28.0

